### ISSUE TITLE
Allow to place the inline toolbox above the selection, instead of below💡

### ISSUE THREAD

**Issue by maxeckel** [codex-team/editor.js#1501]
1. Describe the problem.

When having an extension like "Grammarly" active, its popup will overlap the current selection within the editor.
This causes a quite annoying writing experience, while it actually should become better because of this kind of plugins.

2. Describe the solution you'd like. Mockups are welcome.

I see that this is not necessarily a problem of Editor.js, but it can be "easily" solved by it.
I'd suggest, that the inline toolbox position can be configured to be above the selection instead of below.
Here is a screenshot, which displays the issue:

![image](https://user-images.githubusercontent.com/4929928/104122493-89855e80-5345-11eb-80ec-b97462f3902d.png)

3. Are there any alternatives?

As an alternative one could disable the Grammarly extension for the Editor, but that would drastically lower the writing experience for users and also could be the source of confusion because the users won't understand why Grammarly won't be available in that context.

Thank you for this awesome editor and all the hard work you've put into it!

**Comment by lhapaipai** [comment]
I totally agree with you. another reason for that: if you select a single word, the tooltip will hide the result. (for example if you want to bold)

![bold](https://user-images.githubusercontent.com/1088155/105066711-5ba7c480-5a7f-11eb-8084-328364759309.gif)

**Comment by williamhakim10** [comment]
For those out there looking at this, I was able to hack this by styling the inline toolbar:

```
.ce-inline-toolbar--showed {
  transform: translateY(-65px) translateX(-50%);
}
```

I didn't test this extensively, but seems to work fine. It doesn't quite preserve the location of the inline toolbar (it's always centered, rather than sometimes left-aligned) but I'm sure with a bit more hacking around you could fix that if it's important.

**Comment by imransilvake** [comment]
any solution to this?

**Comment by Oyshoboy** [comment]
same problem here, any non-custom solution?

**Comment by neSpecc** [comment]
It is placed below the selection because, on mobile devices, a native formatting toolbar appears above. We might consider having separate behaviors for mobile and desktop

**Comment by domingoladron** [comment]
I tried css styling, but it was just not precise enough.

Instead, you can set an onmouseup event on the editorjs configuration you pass to find the cursor bounding rectangle and move the inline toolbar closer to that.

```typescript
import { EditorConfig } from "@editorjs/editorjs";

let Configuration = new EditorConfig();
Configuration.onReady = () => {
                const contentEditable = document.getElementById(editorBaseElement.id); //<-- this would be the id of the element to which your Editor will bind.
                contentEditable.onmouseup = (event) => {
                    // we want the inline toolbar to stay close to the action.

                     // if we can't find an exact selection, then you could do nothing, just pop the editor inlineToolbar etc
                     if (window.getSelection().rangeCount <= 0) {
                         return;
                     }

                    const range = window.getSelection().getRangeAt(0);
                    const rect = range.getBoundingClientRect();

                    const contentEditableRect = contentEditable.getBoundingClientRect();
                    const x = rect.left - contentEditableRect.left + contentEditable.scrollLeft;
                    const y = rect.top - contentEditableRect.top + contentEditable.scrollTop;
                    const editorInlineToolbar = document.getElementsByClassName('ce-inline-toolbar').item(0);

                    if (editorInlineToolbar) {
                        const htmlElement = editorInlineToolbar as HTMLElement;
                        htmlElement.style.left = `${x + 5}px`;
                        htmlElement.style.top = `${y + 5}px`;
                        htmlElement.style.display = 'block';
                    }
                };

const editor = new EditorJS(Configuration);

```

Anyone wants, I can whip up a proper TS class which encapsulates this logic if you don't quite get it / need help.  It's a bit rough around the edges, but it basically works.
